src/main/kotlin/rhmodding/bread/editor/SpritesTab.kt

```diff
@@ -127,9 +127,38 @@ open class SpritesTab<F : IDataModel>(editor: Editor<F>) : EditorSubTab<F>(edito
                                 alert.title = "Remove this sprite?"
                                 alert.headerText = "Remove this sprite?"
                                 alert.contentText = "Are you sure you want to remove this sprite?\nYou won't be able to undo this action."
+
                                 if (alert.showAndWait().get() == ButtonType.OK) {
-                                    editor.removeSprite(currentSprite)
-                                    updateSpriteSpinners(false)
+                                    // Check every step because at this point we don't know if the sprite is used in any anims
+                                    var cancel = false
+                                    val index = data.sprites.indexOf(currentSprite).toUShort()
+                                    for (anim in data.animations) {
+                                        var isbreak = true
```

Comment:
Should be `false`, otherwise the for loop breaks after checking the first animation and the second dialogue box doesn't appear regardless of the sprite being present in an animation or not (sprite is still deleted)

Reply:
how did i not notice that XD